src/main/java/io/kestra/plugin/fs/realtime/RealtimeTrigger.java

```diff
@@ -0,0 +1,79 @@
+package io.kestra.plugin.fs.realtime;
+
+import io.kestra.core.models.conditions.ConditionContext;
+import io.kestra.core.models.executions.Execution;
+import io.kestra.core.models.tasks.Output;
+import io.kestra.core.models.triggers.*;
+import io.kestra.core.queues.QueueException;
+import io.kestra.core.queues.QueueFactoryInterface;
+import io.kestra.core.queues.QueueInterface;
+import io.kestra.core.runners.RunContext;
+import jakarta.inject.Inject;
+import jakarta.inject.Named;
+import lombok.NoArgsConstructor;
+import lombok.experimental.SuperBuilder;
+
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
+@SuperBuilder
+@NoArgsConstructor
+public abstract class RealtimeTrigger<T extends Output>
+    extends AbstractTrigger implements RealtimeTriggerInterface {
+
+    private transient ExecutorService executor;
+
+    @Inject
+    @Named(QueueFactoryInterface.EXECUTION_NAMED)
+    private QueueInterface<Execution> executionQueue;
+
+    public void start(RunContext runContext, TriggerContext triggerContext, ConditionContext conditionContext) throws Exception {
+        if (executor == null || executor.isShutdown()) {
+            executor = Executors.newSingleThreadExecutor();
+        }
+
+        executor.submit(() -> {
```

Comment:
You start a single-threaded executor here, which is good, but in the concrete TCP trigger you're spawning manual threads again (`new Thread(...).start()`).
To keep thread management consistent, all background work should be delegated to this executor.
That way, `stop()` can cleanly shut everything down and you avoid orphan threads.
See Debezium plugin's [AbstractDebeziumRealtimeTrigger](https://github.com/kestra-io/plugin-debezium/blob/master/plugin-debezium/src/main/java/io/kestra/plugin/debezium/AbstractDebeziumRealtimeTrigger.java) for reference.